### DISCUSSION TITLE
Action Group

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#234] [Component suggestions]
<img width="480" alt="visuele weergave van het action group component" src="https://github.com/nl-design-system/backlog/assets/545207/6b73a212-2c20-4d71-b37f-f4f085465e44">

We willen graag meer leren over het **Action Group** component.

- In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?
- Zijn er specifieke richtlijnen voor positie of volgorde van de acties?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1711356638851909)

---

## Naam
Action Group

Update: Naam aangepast van 'Button Group' naar 'Action Group'.
[Bekijk de vraag die we stelden over Button Group vs. Action Group op Slack.](https://codefornl.slack.com/archives/C7KDQRXMZ/p1718088742075159?thread_ts=1716789733.101899&cid=C7KDQRXMZ)

## Doel
Groepeert één of meer gerelateerde acties en verzorgt de lay-out van de content.

## Gerelateerde componenten
[Button](https://github.com/orgs/nl-design-system/discussions/205)

## Onderzoek

[Buttons on the web: placement and order- Medium](https://uxdesign.cc/buttons-placement-and-order-bb1c4abadfcb)
Artikel waarin Artem Syzonenko inzichten uit zijn desk research naar de positie en volgorde van acties beschrijft.

[Button Ambiguity: Alignment & Order - Medium](https://medium.theoremone.co/button-ambiguity-alignment-order-a42736e25334)
Artikel waarin Jon Lehman inzichten uit zijn onderzoek naar de positie en volgorde van acties beschrijft.

[Bekijk het Button component voor meer informatie](https://github.com/orgs/nl-design-system/discussions/205)

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-layout-action-group--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/action-group)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9796-63359&mode=design&t=LGCNWEUCOjCsqcVS-4) (als onderdeel van Button component)

---

### Screenshots / Varianten

**Figma**

Button group
![Scherm­afbeelding 2024-03-25 om 11 09 26](https://github.com/nl-design-system/backlog/assets/545207/ed14d9ad-c82e-4fa6-8d7a-aea4a5dc5a57)

In modal dialog
![Scherm­afbeelding 2024-03-25 om 11 18 39](https://github.com/nl-design-system/backlog/assets/545207/443042ff-1a38-4fed-9bd4-4de02ee06884)

---

### Notities
- Variants per viewport.
- Optie voor 'Horizontal' en 'Vertical' uitlijnen.
- Optie voor 2 of 3 acties.
- Horizontal 2 acties van links naar rechts: Tertiary + Primary.
- Horizontal 3 acties van links naar rechts: Tertiary + Secondary + Primary.
- Vertical 2 acties van boven naar beneden: Primary + Tertiary.
- Vertical 3 acties van boven naar beneden: Primary + Secondary + Tertiary.
- De groep wordt binnen een Modal dialog rechts uitgelijnd.

[Bekijk het Button component van Amsterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/205#discussioncomment-8165957)

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-input-button-group--docs)
- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/css-feedback-modal--default-story) (in modal dialog)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/ButtonGroup)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=26011-163705&mode=design&t=dW4Ft5V5xRGDdlCt-4) (als onderdeel van Button component)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=3256-9053&mode=design&t=EnnQZmLMBa4784tK-4)

---

### Screenshots / Varianten

**Storybook**

Single
![Scherm­afbeelding 2024-03-25 om 11 23 31](https://github.com/nl-design-system/backlog/assets/545207/cdfcc76f-7e43-4f94-b75c-2e3aa5997d46)

Multiple
![Scherm­afbeelding 2024-03-25 om 11 23 37](https://github.com/nl-design-system/backlog/assets/545207/08461930-0c5e-44b9-b174-fecd9c97b3f2)

In modal dialog
![Scherm­afbeelding 2024-03-25 om 11 41 12](https://github.com/nl-design-system/backlog/assets/545207/80eaaca9-33ae-4476-8feb-e872278af820)


**Figma - Huidig**

Horizontal
![Scherm­afbeelding 2024-03-25 om 11 25 29](https://github.com/nl-design-system/backlog/assets/545207/55efaa0d-68e2-4584-9637-853f99771f84)

![Scherm­afbeelding 2024-03-25 om 11 36 22](https://github.com/nl-design-system/backlog/assets/545207/c3b92c35-8822-48f4-8143-3a3b703eec8d)

Vertical
![Scherm­afbeelding 2024-03-25 om 11 25 37](https://github.com/nl-design-system/backlog/assets/545207/f85fb7f4-f5c7-44ff-92dc-f5557459d3db)

In modal dialog
![Scherm­afbeelding 2024-03-25 om 11 42 52](https://github.com/nl-design-system/backlog/assets/545207/0357ca4f-4d18-44f6-9078-b227cf6a9aac)

![Scherm­afbeelding 2024-03-25 om 11 43 01](https://github.com/nl-design-system/backlog/assets/545207/f941ec2b-c7ab-4c8a-9f0a-41cc1c4a729e)


**Figma - WIP**

Vertical - False
![Scherm­afbeelding 2024-03-25 om 11 27 18](https://github.com/nl-design-system/backlog/assets/545207/fe91adb1-3c2c-46f0-b98f-f254df367496)

Vertical - True
![Scherm­afbeelding 2024-03-25 om 11 27 11](https://github.com/nl-design-system/backlog/assets/545207/3a9da641-ec16-4ff2-9b9a-fb3f400902b9)

Actions - 2
![Scherm­afbeelding 2024-03-25 om 11 27 18](https://github.com/nl-design-system/backlog/assets/545207/1646555a-8912-44e2-813f-875d7c412233)

Actions - 1
![Scherm­afbeelding 2024-03-25 om 11 27 23](https://github.com/nl-design-system/backlog/assets/545207/6415ee27-0447-4d85-8075-8d29ea416490)

---

### Notities

Storybook
- Optie voor 1 of 2 acties.
- Acties van links naar rechts: Secondary + Secondary (in Modal dialog Primary + Secondary).
- Button group wordt binnen een Modal dialog rechts uitgelijnd.

Figma - Huidig
- Optie voor 'Horizontal' en 'Vertical' uitlijnen.
- Optie voor 1 of 2 acties.
- Horizontaal 2 acties van links naar rechts: Primary + Secondary of Secondary + Primary.
- Verticaal 2 acties van boven naar beneden: Primary + Secondary.
- Button group wordt binnen een Modal dialog rechts uitgelijnd.
- Komt ook voor in Cookie banner en Task component.

Figma - WIP
- Optie voor 'Horizontal' en 'Vertical' uitlijnen.
- Optie voor 1 of 2 acties.
- Horizontaal 2 acties van links naar rechts: Primary + Secondary.
- Verticaal 2 acties van boven naar beneden: Primary + Secondary.
- Bij verticaal uitlijnen worden de buttons uitgevuld over gehele breedte.
- De groep wordt binnen een Modal dialog links uitgelijnd.
- Komt ook voor in Alert component.

[Bekijk het Button component van Den Haag voor meer informatie](https://github.com/orgs/nl-design-system/discussions/205#discussioncomment-8166360)

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-button-group--docs)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=16268-10877&mode=design&t=ZoRRQKdHcB9RgeLT-4) (in modal dialog)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-25 om 12 12 58](https://github.com/nl-design-system/backlog/assets/545207/7ef3f821-8fb9-4438-aa99-d18ca00fb482)

**Figma**
In modal dialog
![Scherm­afbeelding 2024-03-25 om 12 12 26](https://github.com/nl-design-system/backlog/assets/545207/9c1d20d4-cc70-41da-93d4-bb8c6491f7a6)


---

### Notities
Storybook
- Enkel horizontaal uitlijnen.
- 4 acties.
- Van links naar rechts: Primary + Secondary (zwart) + Secondary (groen) + Text button

Figma (in modal dialog)
- Horizontaal en verticaal uitlijnen.
- Horizontaal 2 acties van links naar rechts: Tertiary + Primary.
- Verticaal 2 acties van boven naar beneden: Primary + Tertiary.
- Bij verticaal uitlijnen worden de buttons uitgevuld over gehele breedte.
- De groep wordt binnen een Modal dialog rechts uitgelijnd.

[Bekijk het Button component van Rotterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/205#discussioncomment-8167372)

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-action-group--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/action-group)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4207-18737&mode=design&t=Ih4kzQIj7EWgXk9X-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-25 om 12 30 30](https://github.com/nl-design-system/backlog/assets/545207/d67429b9-7b3b-490a-80fc-7038af6e46e2)

Full width = true
![Scherm­afbeelding 2024-03-25 om 12 32 49](https://github.com/nl-design-system/backlog/assets/545207/e8aa6b13-7029-432f-958a-c968c35c4b45)


---

### Notities
- In Storybook lijken veel verschillende combinaties mogelijk te zijn. Screenshot is default state.
- Horizontaal 2 acties van links naar rechts: Secondary + Primary.
- Horizontaal justify-content: space-between.
- Verticaal (Full width = true) 2 acties van boven naar beneden: Primary + Secondary.
- Full width = true zorgt logische wijs dat buttons uitgevuld worden over gehele breedte.
- Zie ook ['Verwijderen' en 'Wijzigen' in een table row](https://nl-design-system.github.io/rvo/?path=/story/demo-pagina-s-jonge-vissers-kosten--default).

[Bekijk het Button component van RVO voor meer informatie](https://github.com/orgs/nl-design-system/discussions/205#discussioncomment-8167473)

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-action-group--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/action-group)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=2969-32582&mode=design&t=kJuvMq0Wp63kbPvc-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-25 om 12 52 49](https://github.com/nl-design-system/backlog/assets/545207/0898ab7a-5728-4d06-9db6-de964df7761e)

Direction Column
![Scherm­afbeelding 2024-03-25 om 12 53 05](https://github.com/nl-design-system/backlog/assets/545207/7fdd3cf0-2d6b-44f5-82a7-1d733886e03b)

**Figma**

Vertical - False
![Scherm­afbeelding 2024-03-25 om 12 53 36](https://github.com/nl-design-system/backlog/assets/545207/fa46be15-0a45-4152-ba95-2335c371c955)

Vertical - True
![Scherm­afbeelding 2024-03-25 om 12 53 40](https://github.com/nl-design-system/backlog/assets/545207/2adf91e7-449f-4998-860b-94eee0fbb624)

Actions - 2
![Scherm­afbeelding 2024-03-25 om 12 53 46](https://github.com/nl-design-system/backlog/assets/545207/a41b7e25-3e9d-46ae-b294-6a33c247b076)

Actions - 1
![Scherm­afbeelding 2024-03-25 om 12 53 50](https://github.com/nl-design-system/backlog/assets/545207/00e84198-3109-482e-8bca-fe5a80fe6142)

---

### Notities
Storybook
- Horizontaal 2 acties van links naar rechts: Primary + Secondary.
- Verticaal 2 acties van boven naar beneden: Primary + Secondary.

Figma
- Optie voor 'Horizontal' en 'Vertical' uitlijnen.
- Optie voor 1 of 2 acties.
- Horizontaal 2 acties van links naar rechts: Primary + Secondary.
- Verticaal 2 acties van boven naar beneden: Primary + Secondary.
- Bij verticaal uitlijnen worden de buttons uitgevuld over gehele breedte.
- De groep wordt binnen een Modal dialog links uitgelijnd.
- Komt ook voor in Alert component.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨


Help je mee hem door de Community stappen te krijgen?
[Action group op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=action+group)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Action group op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=action+group)

**Comment by jeffreylauwers** [comment]
Wanneer plaats je een call-to-action full-width?
  - Wat is hierbij je afweging?
    - Indien je dit toepast als responsive gedrag, is dit dan altijd hetzelfde gedrag of is het 'instelbaar'? En heeft het te maken met een 'Small viewport' of eerder een small container (Modal Dialog,f Drawer, etc.)

Instelbaar als in:

- Stacked: True/False
- Strech: True/False

<img width="1400" height="1268" alt="Ontwerp toont verschillende toepassingen van de action groep bij een kleine en grote container. Bij de eerste variant staan de buttons boven elkaar en hebben ze een breedte op basis van hun tekst-label. Bij de tweede variant staan de buttons naast elkaar en hebben ze een breedte op basis van hun tekst-label. Bij de derde variant staan button naast elkaar en vullen ze de beschikbare ruimte op. Bij de vierde variant staan de button boven elkaar en en vullen ze de beschikbare ruimte op." src="https://github.com/user-attachments/assets/8385fce8-0cbc-466f-9b8e-6e8f02052ef6" />

Verschilt je aanpak bij 1 call to action? En wat als de andere call-to-action in de groep geen secondary button is, maar bijvoorbeeld 2 linkjes?
